src/ZoneServer/Commands/ChatCommands.Handlers.cs

```diff
@@ -327,6 +330,118 @@ private CommandResult HandleJump(Character sender, Character target, string mess
 			return CommandResult.Okay;
 		}
 
+		/// <summary>
+		/// Official slash command to hire a pet
+		/// </summary>
+		/// <example>/pethire 3 Pet</example>
+		/// <param name="sender"></param>
+		/// <param name="target"></param>
+		/// <param name="message"></param>
+		/// <param name="command"></param>
+		/// <param name="args"></param>
+		/// <returns></returns>
+		private CommandResult HandlePetHire(Character sender, Character target, string message, string command, Arguments args)
+		{
+			// Since this command is sent via UI interactions, we'll not
+			// use any automated command result messages, but we'll leave
+			// debug messages for now, in case of unexpected values.
+			if (args.Count < 2)
+			{
+				Log.Debug("HandlePetHire: Invalid call by user '{0}': {1}", sender.Username, command);
+				return CommandResult.Okay;
+			}
+
+			if (sender.Companions.HasCompanions)
+				return CommandResult.Okay;
+
+			if (!int.TryParse(args.Get(0), out var petShopId))
+				return CommandResult.InvalidArgument;
+
+			if (!ZoneServer.Instance.Data.CompanionDb.TryFind(petShopId, out var data))
+				return CommandResult.InvalidArgument;
+
+			if (!ZoneServer.Instance.Data.MonsterDb.TryFind(data.ClassName, out var monData))
+				return CommandResult.InvalidArgument;
+
+			var targetNpcName = "Companion Trader";
```

Comment:
Correct me if I'm wrong, but won't this check fall apart if the NPC used is not explicitly named `Companion Trader`? It seems like this would be a bit of issue. Not only in cases where the admin might rename NPCs, but also when translations are applied.

Reply:
You are correct, but it was an attempt to prevent abuse of the /pethire command without using the "Npc". I can remove the check for now.